src/proof/PGTransformationAlgorithms.cc

```diff
@@ -555,73 +555,64 @@ void ProofGraph::proofTransformAndRestructure(const double left_time, const int
     }
 
     double init_time = cpuTime();
-    assert(!(max_num_loops > 0 && left_time > 0));
+    assert(not (max_num_loops > 0 and left_time > 0));
     //Flag to check if in a loop at least a transformation has been applied
     bool some_transf_done;
     long curr_num_loops = 0;
-    RuleContext ra1, ra2;
     std::deque<clauseid_t> q;
-    clauseid_t id;
     // Main external loop
     do {
+        assert(isResetVisited1() and isResetVisited2());
         // Enqueue leaves first
         q.assign(leaves_ids.begin(), leaves_ids.end());
         some_transf_done = false;
         do {
-            id = q.front();
+            clauseid_t id = q.front();
             assert(id < getGraphSize());
             q.pop_front();
             ProofNode * n = getNode(id);
             //Node not visited yet
-            if (n != nullptr && !isSetVisited1(id)) {
+            if (n and not isSetVisited1(id)) {
                 // Wait if one antecedent is not visited yet;
                 // the node will be enqueued anyway by that antecedent
-                if ((n->getAnt1() != nullptr && !isSetVisited1(n->getAnt1()->getId()))
-                    || (n->getAnt2() != nullptr && !isSetVisited1(n->getAnt2()->getId()))) {
+                if ((n->getAnt1() and not isSetVisited1(n->getAnt1()->getId()))
+                    or (n->getAnt2() and not isSetVisited1(n->getAnt2()->getId()))) {
                     continue;
                 }
                 // ELSE, both antecedents ready, process the node
                 // Mark node as visited
                 setVisited1(id);
                 // Non leaf node
-                if ((n->getAnt1() != nullptr || n->getAnt2() != nullptr)) {
+                if ((n->getAnt1() or n->getAnt2())) {
                     assert(n->getAnt1());
-                    assert (n->getAnt2());
-                    assert((n->getAnt1()->getAnt1() != nullptr && n->getAnt1()->getAnt2() != nullptr)
-                           || (n->getAnt1()->getAnt1() == nullptr && n->getAnt1()->getAnt2() == nullptr));
-                    assert((n->getAnt2()->getAnt1() != nullptr && n->getAnt2()->getAnt2() != nullptr)
-                           || (n->getAnt2()->getAnt1() == nullptr && n->getAnt2()->getAnt2() == nullptr));
-
-                    // NOTE not clear how this might happen
-                    assert(n->getAnt1() != n->getAnt2());
-                    if (n->getAnt1() == n->getAnt2()) {
-                        opensmt_error("Bad proof structure!");
-                    }
+                    assert(n->getAnt2());
+                    assert((n->getAnt1()->getAnt1() and n->getAnt1()->getAnt2())
+                           or (n->getAnt1()->getAnt1() == nullptr and n->getAnt1()->getAnt2() == nullptr));
```

Comment:
`or (not n->getAnt1()->getAnt1() and not n->getAnt1()->getAnt2()));`?

Reply:
Fixed.